src/Plugin/EntityBrowser/Widget/View.php

```diff
@@ -84,23 +85,26 @@ public function getForm(array &$original_form, FormStateInterface $form_state, a
     $form = [];
     // TODO - do we need better error handling for view and view_display (in case
     // either of those is nonexistent or display not of correct type)?
-    /** @var \Drupal\views\ViewExecutable $view */
 
     $form['#attached']['library'] = ['entity_browser/view'];
 
+    /** @var \Drupal\views\ViewExecutable $view */
     $view = $this->entityManager
       ->getStorage('view')
       ->load($this->configuration['view'])
       ->getExecutable();
 
-    // Add exposed filter values, if present
-    foreach ($form_state->getUserInput() as $name => $value) {
-      if (strpos($name, 'entity_browser_exposed_') === 0) {
-        $name = str_replace('entity_browser_exposed_', '', $name);
-        $view->exposed_data[$name] = $value;
-      }
+    // Check if the current user has access to this view.
+    if (!$view->access($this->configuration['view_display'])) {
+      return [
+        '#markup' => t('You do not have access to this View.'),
+      ];
     }
 
+    // Force AJAX as we are embedding a form within a form, and exposed filter
+    // actions will submit the entire EntityBrowserForm form, not the View.
+    $view->setAjaxEnabled(TRUE);
```

Comment:
I was testing with node view that comes with entity_browser_example and it was not working until I specifically enabled ajax. Are we sure that this actually forces AJAX to be enabled?

I was using this view: https://gist.github.com/slashrsm/0922e783c05bd066bd8b

It is basically default view that ships with the example module with added exposed filter and sort. I am wondering if we should include that.